File: WallAI.HostedSimulation.Web/app/Rpc/RpcResponseError.ts
import { RpcResponse } from "./RpcResponse";
import { RpcResponseType } from "./RpcResponseTypes";

export class RpcResponseError extends RpcResponse {
    message: string;

    constructor(id: string, message: string) {
        super(id, RpcResponseType.Error);
        this.message = message;
    }
}

File: WallAI.HostedSimulation/Servers/Rpc/RpcResponseError.cs
using System;
using System.Runtime.Serialization;

namespace WallAI.HostedSimulation.Servers.Rpc
{
    [DataContract]
    public class RpcResponseError : RpcResponse
    {
        [DataMember(Name = "message")]
        public readonly string Message;

        public RpcResponseError(Guid id, string message) : base(id, RpcResponseType.Error)
        {
            Message = message;
        }
    }
}